pytest_mpi/parallel_assert.py

```diff
@@ -19,24 +19,37 @@ def parallel_assert(assertion: Callable, participating: bool = True, msg: str =
     Notes
     -----
     It is very important that ``parallel_assert`` is called collectively on all
-    ranks simulataneously.
+    ranks simultaneously.
+    This function allows passing a callable instead of a boolean for the
+    `assertion` argument. This is useful in rare circumstances, such as when
+    the assertion is not defined on all tasks, but is not recommended.
```

Comment:
I would put this bit in the description of `assertion` above, but wouldn't even bother saying "is useful in rare circumstances".